test/e2e/tuning_options_test.go

```diff
@@ -320,3 +327,170 @@ func TestHAProxyTimeoutsRejection(t *testing.T) {
 		}
 	}
 }
+
+// TestCookieLen verifies that when logging is enabled and a cookie capture is specified, any captured cookies are
+// truncated at the expected max length.
+func TestCookieLen(t *testing.T) {
+	t.Parallel()
+	icName := types.NamespacedName{
+		Namespace: operatorNamespace,
+		Name:      names.SimpleNameGenerator.GenerateName("test-cookielen-"),
+	}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+	cookieName := "X-foo-bar"
+	maxLength := 128
+	ic.Spec.Logging = &operatorv1.IngressControllerLogging{
+		Access: &operatorv1.AccessLogging{
+			Destination: operatorv1.LoggingDestination{
+				Type: "Container",
+			},
+			HTTPCaptureCookies: []operatorv1.IngressControllerCaptureHTTPCookie{{
+				IngressControllerCaptureHTTPCookieUnion: operatorv1.IngressControllerCaptureHTTPCookieUnion{
+					MatchType: operatorv1.CookieMatchTypeExact,
+					Name:      cookieName,
+				},
+				MaxLength: maxLength,
+			}},
+			HttpLogFormat: "cookie: %CC",
+		},
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("Failed to create ingresscontroller %s: %v", icName, err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	if err := waitForIngressControllerCondition(t, kclient, 2*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+		t.Errorf("Timed out waiting for ingresscontroller %s to become available: %v", icName.Name, err)
+	}
+
+	// Send a request through the router that includes the captured cookie, and verify that it's truncated as expected.
+
+	// Use the router image for the client pod since it includes curl.
+	routerDeployment := &appsv1.Deployment{}
+	routerDeploymentName := controller.RouterDeploymentName(ic)
+	if err := kclient.Get(context.TODO(), routerDeploymentName, routerDeployment); err != nil {
+		t.Fatalf("Failed to get router deployment for ingresscontroller %s: %v", icName.Name, err)
+	}
+
+	curlPodImage := routerDeployment.Spec.Template.Spec.Containers[0].Image
```

Comment:
We should first assert on the length of Spec.Containers == 1.

Reply:
Rather than assert on `Spec.Containers = 1` (since if logging is set to "container" on the default ingresscontroller, there would be 2 containers), I updated it to loop through the containers and get the image from the one named "router" and `t.Fatalf` if that fails. Does that satisfy your concern?